subsys/bluetooth/host/gatt.c

```diff
@@ -1114,6 +1114,56 @@ static void db_changed(void)
 #endif
 }
 
+static void gatt_unregister_ccc(struct _bt_gatt_ccc *ccc)
+{
+	ccc->value = 0;
+
+	for (size_t i = 0; i < ARRAY_SIZE(ccc->cfg); i++) {
+		struct bt_gatt_ccc_cfg *cfg = &ccc->cfg[i];
+
+		if (bt_addr_le_cmp(&cfg->peer, BT_ADDR_LE_ANY)) {
+			struct bt_conn *conn;
+			bool store = true;
+
+			conn = bt_conn_lookup_addr_le(cfg->id, &cfg->peer);
+			if (conn) {
+				if (conn->state == BT_CONN_CONNECTED) {
+#if defined(CONFIG_BT_SETTINGS_CCC_STORE_ON_WRITE)
+					gatt_ccc_conn_enqueue(conn);
```

Comment:
Do we really want to delay writing here? Imo since the CCC is no longer present on the database we should actually remove it immediately.

Note: We only use a delayed work because otherwise settings_save_one would be called from RX thread on bt_gatt_attr_write_ccc which should not be the case of bt_gatt_service_unregister.

Reply:
I thought it best since there might be other CCC write operations pending, or arriving afterwards, and this would avoid bt_gatt_settings_unregister from possibly being preempted by the k_delayed_work.

Reply:
Operation arriving afterward would be discarded as the attribute no longer exists, for the existing/pending operations Id guess we should actually cancel them and not reschedule since we know at this point the CCC is gone nothing else should be able to access it.

Reply:
I meant CCC operations on other characteristics, since multiple CCC operations might be bunched up into on e write operation. The characteristic is not in the list anymore, so it won't be accessible anynore